silica/compile.py

```diff
@@ -194,29 +207,38 @@ def compile(coroutine, file_name=None, mux_strategy="one-hot", output='verilog',
             var = name + f"_{value}_i{count}"
             width_table[var] = width
```

Comment:
Might be nice to factor these out into separate functions, I can take care of that, just noting it here.